subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/TestFile.java

```diff
@@ -274,10 +274,16 @@ public void copyFrom(File target) {
     }
 
     public void copyFrom(URL resource) {
```

Comment:
I'd love to see a test case for this method to ensure all situations are handled properly. Mocking or stubbing `FileUtils` might be a bit challenging though.